core/tracker/inclusion.go

```diff
@@ -193,15 +183,12 @@ func (i *inclusionCore) CheckBlock(ctx context.Context, block block) {
 			i.attIncludedFunc(ctx, sub, block)
 			i.trackerInclFunc(sub.Duty, sub.Pubkey, sub.Data, nil)
 			delete(i.submissions, key)
-		case core.DutyProposer, core.DutyBuilderProposer:
+		case core.DutyProposer:
 			if sub.Duty.Slot != block.Slot {
 				continue
 			}
 
 			msg := "Broadcasted block included on-chain"
-			if sub.Duty.Type == core.DutyBuilderProposer {
```

Comment:
Improved this to log different message for blinded blocks. Hope this is sufficient. But we don't track per-duty anymore, that's true.

Reply:
I think we can, since we just have one duty type, if Charon is configured for builder api we’re sure we only deal with blinded blocks and vice versa.

Reply:
Hmm you're right, the API says:

> - `builder_boost_factor=0`: prefer the execution node payload unless an error makes it unviable.
> - `builder_boost_factor=100`: default profit maximization mode; choose whichever payload pays more.
> - `builder_boost_factor=2**64 - 1`: prefer the builder payload unless an error or beacon node health check makes it unviable.

They're using `prefer`, which means that at least in the last case, there might be a situation in which the builder payload is unavailable. The behavior in case this happens appears to be unspecified?

That said, it would be good to at least have the Charon configuration flag for builder API appended in the log line.

Do you think it's worth it?

Reply:
I changed the implementation here already to render "blinded" word depending on the block. Which is almost similar to previous implementation.